lib/shoehorn/application/event_supervisor.ex

```diff
@@ -38,15 +48,36 @@ defmodule Shoehorn.Application.EventSupervisor do
     :ok
   end
 
-  defp shutdown(app, reason, s) do
+  defp exited(app, reason, s) do
     {:ok, shutdown_timer_ref} = :timer.apply_after(s.shutdown_timer, :erlang, :halt, [])
-
-    case apply(s.handler, :handle_application, [reason, app, s.handler_state]) do
-      {:halt, _handler_state} ->
+    try do
+      case apply(s.handler, :application_exited, [app, reason, s.handler_state]) do
+        {:halt, _handler_state} ->
+          :erlang.halt()
+        {:continue, handler_state} ->
+          :timer.cancel(shutdown_timer_ref)
+          %{s | handler_state: handler_state}
+      end
+    rescue
+      e ->
+        IO.puts("Shoehorn handler raised an exception: #{inspect e}")
+        IO.puts("halt")
         :erlang.halt()
-      {:continue, handler_state} ->
-        :timer.cancel(shutdown_timer_ref)
-        %{s | handler_state: handler_state}
+    end
+  end
+
+  defp started(app, s) do
+    try do
+      case apply(s.handler, :application_started, [app, s.handler_state]) do
+        {:halt, _handler_state} ->
+          :erlang.halt()
+        {:continue, handler_state} ->
+          %{s | handler_state: handler_state}
+      end
+    rescue 
```

Comment:
same here

Reply:
Do we want to continue if something fails startup? Will we ever be able to respond to restart it?

Reply:
This is the place where we are receiving a notification that the application _has_ started successfully.

Reply:
Yes, but if our handler isn't stable enough to handle that message I feel like we should assume the integrity of the whole thing as suspect.